policy-controller/k8s/status/src/index.rs

```diff
@@ -74,14 +111,41 @@ impl Index {
         updates: UnboundedSender<Update>,
     ) -> SharedIndex {
         Arc::new(RwLock::new(Self {
-            _name: name.to_string(),
-            _claims: claims,
+            name: name.to_string(),
+            claims,
             updates,
             http_routes: HashMap::new(),
             servers: HashSet::new(),
         }))
     }
 
+    pub async fn run(index: Arc<RwLock<Self>>) {
+        // Clone the claims watch out of the index. This will immediately
+        // drop the read lock on the index so that it is not held for the
+        // lifetime of this function.
+        let mut claims = index.read().claims.clone();
+
+        loop {
+            tokio::select! {
+                _ = claims.changed() => {
+                    tracing::debug!("Lease holder has changed")
+                }
+                _ = tokio::time::sleep(std::time::Duration::from_secs(10)) => {}
```

Comment:
style nit, take it or leave it: `tokio::time` re-exports `Duration`, so we could add a `use tokio::time;` and change this to
```suggestion
                _ = time::sleep(time::Duration::from_secs(10)) => {}
```
which is maybe a little nicer looking? not important though.